File: terraform/backend/pre-token-gen/handler.js
exports.handler = (event, context, callback) => {
  console.log("event", event);
  // event.response = {
  //   claimsOverrideDetails: {
  //     // "claimsToAddOrOverride": {
  //     //     "attribute_key2": "attribute_value2",
  //     //     "attribute_key": "attribute_value"
  //     // },
  //     // "claimsToSuppress": ["email"],
  //     groupOverrideDetails: {
  //       // groupsToOverride: ["system-user"],
  //       // "iamRolesToOverride": ["arn:aws:iam::XXXXXXXXXXXX:role/sns_callerA", "arn:aws:iam::XXXXXXXXX:role/sns_callerB", "arn:aws:iam::XXXXXXXXXX:role/sns_callerC"],
  //       // "preferredRole": "arn:aws:iam::XXXXXXXXXXX:role/sns_caller"
  //     },
  //   },
  // };

  // Return to Amazon Cognito
  callback(null, event);
};

File: terraform/backend/src/pre-token-gen/handler.go
package main

import (
	"fmt"

	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-lambda-go/lambda"
)

func handler(event events.CognitoEventUserPoolsPreTokenGen) (events.CognitoEventUserPoolsPreTokenGen, error) {
	fmt.Printf("PreSignup of user: %s\n", event.UserName)
	fmt.Printf("event %+v\n", event)

	// event.Response.ClaimsOverrideDetails = events.ClaimsOverrideDetails{
	// 	GroupOverrideDetails: events.GroupConfiguration{
	// 		GroupsToOverride: []string{"system-user"},
	// 	},
	// }

	return event, nil
}

func main() {
	lambda.Start(handler)
}
